contracts/contracts/UserReplicaSetManager.sol

```diff
@@ -133,4 +155,76 @@ contract UserReplicaSetManager is RegistryContract, SigningLogic {
       {
           return spIdToCreatorNodeDelegateWallet[_spID];
       }
+
+      /* EIP712 - Signature generation */
+      function generateAddOrUpdateCreatorNodeRequestSchemaHash(
+          uint _cnodeId,
+          address _cnodeWallet,
+          uint _proposerId,
+          bytes32 _nonce
+      ) internal view returns (bytes32)
+      {
+          return generateSchemaHash(
+              keccak256(
+                  abi.encode(
+                      ADD_UPDATE_CNODE_REQUEST_TYPEHASH,
+                      _cnodeId,
+                      _cnodeWallet,
+                      _proposerId,
+                      _nonce
+                  )
+              )
+          );
+      }
+
+      function _generateUpdateReplicaSetRequestSchemaHash(
+          uint _userId,
+          uint _primary,
+          uint[] memory _secondaries,
+          uint _oldPrimary,
+          uint[] memory _oldSecondaries,
+          bytes32 _nonce,
+          bytes memory _subjectSig
+      ) internal returns (address)
+      {
+        bytes32 signatureDigest = generateSchemaHash(
+            keccak256(
+              abi.encode(
+                  UPDATE_REPLICA_SET_REQUEST_TYPEHASH,
+                  _userId,
+                  _primary,
+                  keccak256(abi.encode(_secondaries)),
+                  _oldPrimary,
+                  keccak256(abi.encode(_oldSecondaries)),
+                  _nonce
+              )
+            )
+        );
+        address signer = recoverSigner(signatureDigest, _subjectSig);
+        burnSignatureDigest(signatureDigest, signer);
+        return signer;
+      }
```

Comment:
we should be consistent across thes two generateRequstSchemeHash functions - if you recover and return signer inside this function you should do it in generateAddorUpdate func above as well

Reply:
oh yes, definitely agree here